.github/workflows/build.yml

```diff
@@ -115,3 +115,15 @@ jobs:
         uses: golangci/golangci-lint-action@0ad9a0988b3973e851ab0a07adf248ec2e100376 # v3.3.1
         with:
           version: ${{ env.GOLANGCI_LINT_VERSION }}
+
+  kernel-tests:
+    name: Kernel testing
+    runs-on: ubuntu-latest
+    needs: go-build-test
+    steps:
+        - name: Install qemu and curl
+          run: sudo apt -y install qemu-system-x86 curl
```

Comment:
To note, our previous experiences with QEMU were sub-par. It was painfully slow. Mainly emulating ARM on these action machines. That being said, we can always try to run these when we merge to the main, and it'd reduce the pain on PRs. Let's see how this goes.

Reply:
Indeed! That's something I want to be mindful of. I think that the past experiences were particularly bad because we were emulating arm64. On x86 this shouldn't be as slow, despite [GH actions not supporting nested virtualization](https://github.com/community/community/discussions/8305).

Either way, I suggest that we first see how long it would take, and then decide on the course of action. For example, we could do the VM tests periodically. Something like every day? That way we can ensure that the profiler works fine in different kernels for most commits in `main` and there shouldn't be that many changes to bisect if something breaks.

If we go for this approach we need to decide how is the responsibility of troubleshooting breakages going to be distributed among the maintainers/team members.

Reply:
I would be happy to experiment. If they take too much time, we can move on to the daily approach and make the GH action open an issue in case of a failure. It should be straightforward to implement. 

Referring to my previous comment, let's separate the action so that we can track how long it takes quickly.

Reply:
Cool, will create another workflow. Hopefully we should be able to do that 😄